inc/ZipProvider.php

```diff
@@ -118,49 +113,25 @@ public function generate_zip_file() : bool {
 			$wp_filesystem->mkdir( static::get_cache_dir(), FS_CHMOD_DIR );
 		}
 
-		/* @var GP_Locale $locale */
-		$locale     = GP_Locales::by_slug( $this->translation_set->locale );
-		$gp_project = GP::$project->get( $this->translation_set->project_id );
-		$entries    = GP::$translation->for_export( $gp_project, $this->translation_set, [ 'status' => 'current' ] );
-
-		if ( ! $entries ) {
-			return false;
-		}
-
-		$files_for_zip = [];
-		$project       = new Project( $gp_project );
-		$text_domain   = $project->get_text_domain();
+		$export = new Export( $this->translation_set );
 
-		/* @var GP_Format $format */
-		foreach ( [ GP::$formats['po'], GP::$formats['mo'] ] as $format ) {
-			$file_name = str_replace( '.zip', '.' . $format->extension, $this->get_zip_filename() );
+		$files_for_zip = $export->export_strings();
 
-			if ( $text_domain ) {
-				$file_name = $text_domain . '.' . $format->extension;
-			}
-
-			$temp_file = wp_tempnam( $file_name );
-
-			$contents = $format->print_exported_file( $gp_project, $locale, $this->translation_set, $entries );
-
-			$wp_filesystem->put_contents( $temp_file, $contents, FS_CHMOD_FILE );
-
-			$files_for_zip[ $temp_file ] = $file_name;
+		if ( ! $files_for_zip ) {
+			return false;
 		}
 
 		$zip = new ZipArchive();
 
 		if ( $zip->open( $this->get_zip_path(), ZipArchive::CREATE ) === true ) {
-			foreach ( $files_for_zip as $temp_file => $file_name ) {
+			foreach ( $files_for_zip as $file_name => $temp_file ) {
 				$zip->addFile( $temp_file, $file_name );
 			}
 
 			$zip->close();
 		}
 
-		foreach ( $files_for_zip as $temp_file => $file_name ) {
-			unlink( $temp_file );
-		}
+		array_map( 'unlink', $files_for_zip );
```

Comment:
Since the files were created with WP_Filesystem, shouldn't they be deleted with the API as well?